go/tasks/plugins/array/awsbatch/transformer.go

```diff
@@ -85,14 +87,22 @@ func FlyteTaskToBatchInput(ctx context.Context, tCtx pluginCore.TaskExecutionCon
 	}
 	resources := flytek8s.ApplyResourceOverrides(*res, *platformResources, assignResources)
 
-	return &batch.SubmitJobInput{
+	submitJobInput := &batch.SubmitJobInput{
 		JobName:            refStr(tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName()),
 		JobDefinition:      refStr(jobDefinition),
 		JobQueue:           refStr(jobConfig.DynamicTaskQueue),
 		RetryStrategy:      toRetryStrategy(ctx, toBackoffLimit(taskTemplate.Metadata), cfg.MinRetries, cfg.MaxRetries),
 		ContainerOverrides: toContainerOverrides(ctx, append(cmd, args...), &resources, envVars),
 		Timeout:            toTimeout(taskTemplate.Metadata.GetTimeout(), cfg.DefaultTimeOut.Duration),
-	}, nil
+	}
+	if taskTemplate.GetCustom() != nil {
+		err = utils.UnmarshalStructToObj(taskTemplate.GetCustom(), &submitJobInput)
```

Comment:
This is a UX question: should we unmarshal first then override with the things above?
```
@task(config=AWSBatch(jobName="blah"))
def task():
  ...
```

Should this use that jobName as opposed to the generated name? I don't think so... 

Then this:
```
@task(config=AWSBatch(retries=aws.RetryStrategy(count=3)),
            retries=2)
def task():
  ...
```

Which one should win here? not sure... we maybe should fail?

I want to say let's unmarshal first the override with the properties we computed... we can add code that says things like: "If retry is set in both, log a warning"

Reply:
The SubmitJobInput properties we computed in the propeller are not included in the [flytekit aws batch config](https://github.com/flyteorg/flytekit/pull/782/files#diff-6126197e8a9fe3df591b719012cd0ef2c32cb87fc027d9d68d5aac7e9c5835abR21-R27)  except `RetryStrategy` and `Timeout`.  
Do we still need to unmarshal first? I think it's fine to override these two properties by flytekit.
or Do we need to also remove RetryStrategy and Timeout in flytekit

Reply:
aha... I would say let's remove them from flytekit then... and keep the common properties consistent with the other flyte tasks...

Reply:
done

Reply:
Let's revert ordering here to make this protocol strict... even if we, in some future, added these properties to flytekit, they should get overriden here... or maybe somebody produced the task outside of flytekit... or maybe flytekit-java doesn't do it correctly... etc. etc... let's keep this protocol tight